Dockerfile

```diff
@@ -1,9 +1,13 @@
-FROM golang:1.20 as builder
-WORKDIR /netreap
-COPY . /netreap
+FROM --platform=${BUILDPLATFORM} golang:1.20.5 as builder
+WORKDIR /app
+COPY go.mod go.sum ./
+RUN go mod download
+COPY . ./
 ARG VERSION
-RUN go build -ldflags "-s -w -X 'main.Version=$VERSION'"
-
-FROM gcr.io/distroless/base-debian11
-COPY --from=builder /netreap/netreap /usr/bin/netreap
-ENTRYPOINT ["/usr/bin/netreap"]
+ARG TARGETOS
+ARG TARGETARCH
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -ldflags "-s -w -X 'main.Version=$VERSION'" -o /netreap
+FROM scratch AS bin
```

Comment:
I'd prefer to stick with the distroless image since it comes with cacerts pre-installed. If you want to stick with a static build without CGO, you can switch to use `gcr.io/distroless/static` instead